app/scripts/modules/core/config/settings.ts

```diff
@@ -0,0 +1,159 @@
+interface IAzureProviderSettings {
+  defaults: {
+    account: string;
+    region: string;
+  };
+}
+
+interface IAWSProviderSettings {
+  defaults: {
+    account: string;
+    region: string;
+    iamRole: string;
+  };
+  defaultSecurityGroups: string[];
+  loadBalancers: {
+    inferInternalFlagFromSubnet: boolean;
+  };
+  useAmiBlockDeviceMappings: boolean;
+}
+
+interface IGCEProviderSettings {
+  defaults: {
+    account: string;
+    region: string;
+    zone: string;
+  };
+}
+
+interface ITitusProviderSettings {
+  defaults: {
+    account: string;
+    region: string;
+    iamProfile: string;
+  };
+}
+
+interface IOpenStackProviderSettings {
+  defaults: {
+    account: string;
+    region: string;
+  };
+}
+
+interface IKubernetesProviderSettings {
+  defaults: {
+    account: string;
+    namespace: string;
+    proxy: string;
+  };
+}
+
+interface IAppEngineProviderSettings {
```

Comment:
I might not be 100% consistent, but I try to not camel case App Engine as one word, i.e., `Appengine`, since the API does the same.